org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/batch/ClasspathJar.java

```diff
@@ -89,7 +89,7 @@ public List<Classpath> fetchLinkedJars(FileSystem.ClasspathSectionProblemReporte
 				directoryPath = directoryPath.substring(0, lastSeparator + 1); // potentially empty (see bug 214731)
 				while (calledFilesIterator.hasNext()) {
 					File linkedFile = new File(directoryPath + (String) calledFilesIterator.next());
-					if (linkedFile.isFile()) {
+					if (linkedFile.isFile() && linkedFile.getName().toLowerCase().endsWith(".jar")) { //$NON-NLS-1$
```

Comment:
If s.o. on this planet every put a jar on their classpath that doesn't have the `.jar` file extension, then this change will be a breaking change for them.
I don't recall JLS requiring that file extension.